packages/widget/src/components/network-selector/network-selector.ts

```diff
@@ -16,70 +17,91 @@ type Direction = (typeof Directions)[keyof typeof Directions];
 export class NetworkSelector extends LitElement {
   static styles = styles;
 
-  @property({
-    type: Boolean
-  })
-  disabled = false;
+  @state()
+  _isDropdownOpen = false;
 
-  @property({
-    type: Boolean
-  })
-  icons = true;
+  @property({ type: Boolean })
+  disabled = false;
 
-  @property({
-    type: String
-  })
+  @property({ type: String })
   direction?: Direction;
 
-  @property({
-    type: Object
-  })
-  selected?: Domain;
+  @property({ type: Object })
+  selectedNetwork?: Domain;
 
-  @property({
-    attribute: false
-  })
+  @property({ attribute: false })
   onNetworkSelected?: (network?: Domain) => void;
```

Comment:
should this property be optional?

Reply:
In general yeah, the component without sending back the state will make no sense, so probably we should make it mandatory